File: python/websearch-research.py
#!/usr/bin/env python3
# websearch-research.sh — portiert nach python
# Quelle: shell, OpenClaw@gateway1:scripts/websearch-research.sh
# Erzeugt: 2026-08-23 durch ABSTRACTIONS_MANAGER.py

# Web Search Script: Deep Research für Incidents
# Verwendung: ./websearch-research.py "Beschreibung des Problems"

import os
import sys
import json
import requests
from datetime import datetime

def main():
    QUERY = sys.argv[1] if len(sys.argv) > 1 else None
    OUTPUT_DIR = sys.argv[2] if len(sys.argv) > 2 else "./research"

    if not QUERY:
        print("Verwendung: {} \"Problem Beschreibung\" [OUTPUT_DIR]".format(sys.argv[0]))
        sys.exit(1)

    os.makedirs(OUTPUT_DIR, exist_ok=True)
    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
    OUTPUT_FILE = os.path.join(OUTPUT_DIR, f"incident_{timestamp}.md")

    with open(OUTPUT_FILE, "w") as f:
        f.write("# Incident Research\n")
        f.write(f"Datum: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}\n")
        f.write(f"Query: {QUERY}\n\n")

        # 1. EXA für schnelle Recherche
        f.write("## 1. Schnelle Recherche (EXA)\n")
        try:
            response = requests.post(
                "https://openrouter.ai/api/v1/chat/completions",
                headers={
                    "Authorization": f"Bearer {os.environ['OPENROUTER_API_KEY']}",
                    "Content-Type": "application/json"
                },
                json={
                    "model": "openai/gpt-5.6-terra",
                    "messages": [{"role": "user", "content": QUERY}],
                    "plugins": [{"id": "web", "engine": "exa", "max_results": 5}]
                },
                timeout=30
            )
            response.raise_for_status()
            data = response.json()
            content = data.get("choices", [{}])[0].get("message", {}).get("content", "Keine Ergebnisse")
            f.write(content + "\n")
        except Exception as e:
            f.write(f"Fehler bei der Abfrage: {str(e)}\n")

        f.write("\n---\n")

        # 2. Verifizierte Quellen (Perplexity) falls verfügbar
        f.write("## 2. Verifizierte Fakten (Perplexity)\n")
        try:
            response = requests.post(
                "https://openrouter.ai/api/v1/chat/completions",
                headers={
                    "Authorization": f"Bearer {os.environ['OPENROUTER_API_KEY']}"
                },
                json={
                    "model": "perplexity/sonar:online",
                    "messages": [{"role": "user", "content": f"{QUERY} troubleshooting"}]
                },
                timeout=30
            )
            response.raise_for_status()
            data = response.json()
            content = data.get("choices", [{}])[0].get("message", {}).get("content", "Keine Ergebnisse")
            f.write(content + "\n")
        except Exception as e:
            f.write(f"Fehler bei der Abfrage: {str(e)}\n")

    print(f"Gespeichert in: {OUTPUT_FILE}")

if __name__ == "__main__":
    main()

File: javascript/websearch-research.js
#!/usr/bin/env node
// websearch-research.sh — portiert nach javascript
// Quelle: shell, OpenClaw@gateway1:scripts/websearch-research.sh
// Erzeugt: 2026-08-23 durch ABSTRACTIONS_MANAGER.py

// Web Search Script: Deep Research für Incidents
// Verwendung: node websearch-research.js "Beschreibung des Problems"

const fs = require('fs');
const https = require('https');
const process = require('process');

const QUERY = process.argv[2];
const OUTPUT_DIR = process.argv[3] || './research';

if (!QUERY) {
  console.log('Verwendung: node websearch-research.js "Problem Beschreibung" [OUTPUT_DIR]');
  process.exit(1);
}

// Hilfsfunktion für HTTP-Anfragen
function httpRequest(options, data) {
  return new Promise((resolve, reject) => {
    const req = https.request(options, (res) => {
      let responseData = '';
      res.on('data', chunk => responseData += chunk);
      res.on('end', () => resolve(JSON.parse(responseData)));
    });
    req.on('error', err => reject(err));
    if (data) {
      req.write(data);
    }
    req.end();
  });
}

// Aktuelles Datum im Format YYYYMMDD_HHMMSS
function getTimestamp() {
  const now = new Date();
  const pad = (num) => String(num).padStart(2, '0');
  return `${now.getFullYear()}${pad(now.getMonth()+1)}${pad(now.getDate())}_${pad(now.getHours())}${pad(now.getMinutes())}${pad(now.getSeconds())}`;
}

// Hauptfunktion
async function main() {
  const timestamp = getTimestamp();
  const outputFile = `${OUTPUT_DIR}/incident_${timestamp}.md`;

  // Verzeichnis erstellen
  fs.mkdirSync(OUTPUT_DIR, { recursive: true });

  // Header schreiben
  let content = `# Incident Research\n`;
  content += `Datum: ${new Date().toString()}\n`;
  content += `Query: ${QUERY}\n\n`;

  // 1. EXA für schnelle Recherche
  content += `## 1. Schnelle Recherche (EXA)\n`;
  
  const exaOptions = {
    hostname: 'openrouter.ai',
    path: '/api/v1/chat/completions',
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${process.env.OPENROUTER_API_KEY}`,
      'Content-Type': 'application/json'
    }
  };

  const exaData = JSON.stringify({
    model: "openai/gpt-5.6-terra",
    messages: [{ role: "user", content: QUERY }],
    plugins: [{ id: "web", engine: "exa", max_results: 5 }]
  });

  try {
    const exaResponse = await httpRequest(exaOptions, exaData);
    content += (exaResponse.choices && exaResponse.choices[0] && exaResponse.choices[0].message && exaResponse.choices[0].message.content) || "Keine Ergebnisse";
  } catch (err) {
    content += "Fehler bei der Anfrage";
  }

  content += "\n\n---\n";

  // 2. Verifizierte Quellen (Perplexity)
  content += `## 2. Verifizierte Fakten (Perplexity)\n`;

  const perplexityOptions = {
    hostname: 'openrouter.ai',
    path: '/api/v1/chat/completions',
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${process.env.OPENROUTER_API_KEY}`,
      'Content-Type': 'application/json'
    }
  };

  const perplexityData = JSON.stringify({
    model: "perplexity/sonar:online",
    messages: [{ role: "user", content: `${QUERY} troubleshooting` }]
  });

  try {
    const perplexityResponse = await httpRequest(perplexityOptions, perplexityData);
    content += (perplexityResponse.choices && perplexityResponse.choices[0] && perplexityResponse.choices[0].message && perplexityResponse.choices[0].message.content) || "Keine Ergebnisse";
  } catch (err) {
    content += "Fehler bei der Anfrage";
  }

  // In Datei schreiben
  fs.writeFileSync(outputFile, content);
  console.log(`Gespeichert in: ${outputFile}`);
}

main();
